opba-embedded-starter/src/main/java/de/adorsys/obpa/starter/OpenBankingEmbeddedApplication.java

```diff
@@ -6,7 +6,7 @@
 import de.adorsys.opba.bankingapi.config.EnableBankingApi;
 import de.adorsys.opba.consentapi.config.EnableConsentApi;
 
-@SpringBootApplication
+@SpringBootApplication(scanBasePackages = {})
```

Comment:
`scanBasePackages = {}` is a default value, ideally we need some Configuration class with these annotations, for example:
```
package de.adorsys.obpa.starter.config;

import de.adorsys.opba.bankingapi.config.EnableBankingApi;
import de.adorsys.opba.consentapi.config.EnableConsentApi;

@EnableBankingApi
@EnableConsentApi
public class ApiConfig {
}
```
And OpenBankingEmbeddedApplication will look like:
```
package de.adorsys.obpa.starter;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

import de.adorsys.opba.bankingapi.config.EnableBankingApi;
import de.adorsys.opba.consentapi.config.EnableConsentApi;

@SpringBootApplication(scanBasePackages = {"de.adorsys.obpa.starter.config"})
@SuppressWarnings("checkstyle:HideUtilityClassConstructor") // Springboot starter class is not an utility class
public class OpenBankingEmbeddedApplication {

    public static void main(String[] args) {
        SpringApplication.run(OpenBankingEmbeddedApplication.class, args);
    }

}
```

Reply:
We do not need this, as EnableBankingApi contains the ComponentScan annotation. Means original version was correct.

Reply:
I thing we can also just limit scanning by adding scanBasePackages = {"de.adorsys.obpa.starter"} as we have no additional config class.

both @EnableBankingApi, @EnableConsentApi can directly be applied on top of the Application class. I do not see the difference.

Reply:
[ComponentScan](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/ComponentScan.html) says: "
```
If specific packages are not defined, scanning will occur from the package of the class that declares this annotation
```

[@SpringBootApplication](https://docs.spring.io/spring-boot/docs/current/reference/html/using-spring-boot.html#using-boot-using-springbootapplication-annotation) is equivalent to // @Configuration @EnableAutoConfiguration @ComponentScan.

So there is no need for additional config class for now. Will remove the scanBasePackages from the annotations.

Reply:
Yes and that means if there are nested packages - they will be scanned even if there is no need to.
See this
https://www.endoflineblog.com/spring-best-practices
Point number 4.
`DO NOT: use entire classpath scanning`